packages/x-components/src/x-installer/api/api.types.ts

```diff
@@ -81,7 +81,7 @@ export interface XAPI {
  */
 export type XEventListeners = Partial<
   {
-    [Event in XEvent]: (payload: XEventPayload<Event>, metadata: WireMetadata) => void;
+    [Event in XEvent]: (payload: XEventPayload<Event>, metadata: WireMetadata) => unknown;
```

Comment:
because the callbacks can return info. This info is used in `SnippetCalbacks` component to emit the event `SnippetCallbackExecuted` (https://github.com/empathyco/x/blob/main/packages/x-components/src/components/snippet-callbacks.vue)

This is useful to request info to customer website. For example on the add to cart callback it returns a promise with boolean to know if the add to cart was succesful.

Reply:
Okey. Thanks for the explanation. I'll dive deeper to understand exactly what we're doing there 😅